### PR TITLE
Use proxy=() instead of removed proxy()

### PR DESCRIPTION
**Pull request by a2ikm**
In faraday 1.0+, we can't use `Faraday::Connection#proxy` to set proxy settings. Instead of it, we can use `Faraday::Connection#proxy=`. This method has been introduced since 0.13.0.
https://github.com/lostisland/faraday/commit/d18cc04be5c179b1b22b9dd1004030dd43b0ca58#diff-ec23de93634c9f1330648fecdfd152dfR288-R290

This PR fixes warnings like this with faraday < 1.0:

```
Warning: use of proxy(new_value) to set connection proxy have been DEPRECATED and will be removed in Faraday 1.0
```

In addition to that, this PR configures matrix tests against faraday and ruby.

### CODE REVIEW

File: gemfiles/faraday-0.16/Gemfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+source 'https://rubygems.org'
+
+# Specify your gem's dependencies in sbpayment.gemspec
+gemspec path: File.expand_path('../../..', __FILE__)
+
+gem 'faraday', '~> 0.16.0'
```
Comment (kachick) [inline gemfiles/faraday-0.16/Gemfile:6]: Should we use fixed versioning?

```suggestion
gem 'faraday', '0.16.0'
```

Comment (a2ikm) [reply gemfiles/faraday-0.16/Gemfile:6]: I didn't care about the patch version, but it sounds good.

**Comment by a2ikm** [comment]
Thank you for your approval!

**Comment by a2ikm** [comment]
Released in v0.11.1 🎉 
https://rubygems.org/gems/sbpayment/versions/0.11.1

**Pull request merged**